### PR TITLE
HPA with custom metrics from Prometheus

### PR DESCRIPTION
**Pull request by kforeverisback**
# Type of PR

- [X] Documentation changes
- [X] Code changes
- [X] Test changes

## Purpose of PR
Add custom metrics capability to HPA using prometheus adapter.

- Automatically deploys prometheus and prometheus-adapter
- Configures all applications for metrics collection
- Creates a new metrics called `NgsaRequestsPerSecond`
- Setup prometheus port and adapter configuration
- Fix latest kubernetes (k3s implementation) metrics deployment issue
  - Latest k3d automatically deploys metrics server
  - Also updated the components.yaml file in case it needs to be deployed
- Added two addons for devexp

## Does this introduce a breaking change

- [ ] YES
- [X] NO

## Validation

- [X] Tests updated and ran successfully
- [X] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/607
- References https://github.com/retaildevcrews/wcnp/issues/596

### CODE REVIEW

File: deploy/prometheus/prom-adapter-helm-values.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        kubernetes_pod_name:
+          resource: pod
+        # deployment:
+        #   resource: "deployment"
+    name:
+      matches: "^(.*)AppDuration_count$" # 1st capture group is ${1} in 'as:' section
+      as: "${1}RequestsPerSecond"
+    metricsQuery: "(sum(rate(<<.Series>>{<<.LabelMatchers>>}[1m])) by (<<.GroupBy>>))"
+    # metricsQuery: 'NgsaAppDuration_count{code="OK",mode!=""}'
+    # metricsQuery: "sum(rate(<<.Series>>{<<.LabelMatchers>>}[30s])) by (<<.GroupBy>>)"
```
Comment (sivamu) [inline deploy/prometheus/prom-adapter-helm-values.yaml:21]: still need commented lines?

**Review by sivamu — APPROVED** [review]
code's good overall, few minor changes needed

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -37,6 +38,8 @@
 		"humao.rest-client",
 		"serayuzgur.crates",
 		"visualstudioexptteam.vscodeintellicode",
+		"oderwat.indent-rainbow",
```
[the remark this answers is not in the document]
Comment (kforeverisback) [reply .devcontainer/devcontainer.json:41]: I found it helpful when modifying multi level yaml files, like our filter.yaml and config files.
image

I can remove it if it's not necessary

File: deploy/prometheus/prom-adapter-helm-values.yaml (hunk 1 above)
Comment (kforeverisback) [reply deploy/prometheus/prom-adapter-helm-values.yaml:21]: It was there mostly for example purposes. I'll remove the comments

Comment (kforeverisback) [reply deploy/prometheus/prom-adapter-helm-values.yaml]: Removed comments

**Comment by kforeverisback** [comment]
@sivamu will merge by EOD today

**Pull request merged**